File: Bobby-PB002/Wikipedia (2)/pb002-2.rb
require 'net/http'
require 'json'
def searcher(tosearch,tostore)
	response = Net::HTTP.get_response(URI.parse("https://en.wikipedia.org/w/api.php?action=opensearch&search="+tosearch+"&limit=1&format=json"))
	data=response.body
	data=JSON.parse(data)
	puts "\nTitle of the page you had requested : "+data[1][0]
	puts "\nUrl of the page you had requested : "+data[3][0]
return data[3][0]
end
if ARGV.length>=2
	tosearch=ARGV.second.to_s
	tostore=ARGV.first.to_s
elsif ARGV.length==1
	tostore=ARGV.first
	puts "Please Enter a string to be searched\nover wikipedia"
	tosearch=STDIN.gets
else
	puts "Please Enter where you want to be store log"
	tostore=STDIN.gets
	puts "Please Enter a string to be searched\nover wikipedia"
	tosearch=STDIN.gets
end

f=File.open(tostore, 'w')
data=searcher(tosearch,tostore)
f.write(tosearch+"resulted to "+data+"\n") 
puts "Press one of the following:\n1 to search again\n2 to exit:"
check=STDIN.gets.to_i
while(check!=2)
puts "Please Enter a string to be searched\nover wikipedia"
tosearch=STDIN.gets

data=searcher(tosearch,tostore)
f.write(tosearch+"resulted to "+data+"\n") 
puts "Press one of the following:\n1 to search again\n2 to exit:"
check=STDIN.gets.to_i
end

File: Bobby-PB002/Wikipedia (2)/pb002-2.py
import sys
import requests
import json 
def searcher(tosearch,tosotore):
	try:
 		wikistr=requests.get("https://en.wikipedia.org/w/api.php?action=opensearch&search="+tosearch+"&limit=1&format=json")
		if(wikistr.status_code==200):
			content=wikistr.content
			o = json.loads(content)
			print "The title of your page is "+o[1][0]
			print "The link of your page is "+o[3][0]
			tosotore.write(tosearch+" rendered "+o[3][0])
			tosotore.write("\n")
		else:
			print "There seems to be an issue in contacting the server"
	except requests.ConnectionError, e:
		print "There seems to be an issue in connecting to the server"
	return(0);
if len(sys.argv)>=2:
	if len(sys.argv)>=3:
		tosotore=sys.argv[1]
	else:
		tosearch=raw_input("Please Enter the string that you want to search : ")
	tosotore=sys.argv[1]
else:
	tosotore=raw_input("Please enter the name of the log file:")
	tosearch=raw_input("Please Enter the string that you want to search : ")
try:
	f=open(tosotore,"w")
except Exception, e:
	"It seems we cant access your file"
	exit()
searcher(tosearch,f)
cont=raw_input("Press one of the following:\n1 to search again\n2 to exit\n:")
while(cont!='2'):
	tosearch=raw_input("Please Enter the string that you want to search : ")
	searcher(tosearch,f)
	cont=raw_input("Press one of the following:\n1 to search again\n2 to exit\n:")	
	exit()
